### PR TITLE
Adding symbol lookup

### PR DESCRIPTION
**Pull request by mat-mcloughlin**

### CODE REVIEW

File: keymaps/omnisharp-atom.cson [hunk 1]
```diff
@@ -33,6 +33,7 @@
   'ctrl-alt-down': 'omnisharp-atom:navigate-down'
   'ctrl-.': 'omnisharp-atom:get-code-actions'
   'alt-enter': 'omnisharp-atom:get-code-actions'
+  'ctrl-shift-t': 'omnisharp-atom:find-symbols'
```
Comment (AndersNS) [inline keymaps/omnisharp-atom.cson]: This is already used by atom. Consult this: https://github.com/atom/atom/tree/master/keymaps :)

File: lib/omnisharp-atom/features/find-symbols.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import Omni = require('../../omni-sharp-server/omni')
+import ClientManager = require('../../omni-sharp-server/client-manager');
+import OmniSharpAtom = require('../omnisharp-atom')
+import SpacePen = require('atom-space-pen-views');
+import FindSymbolsView = require('../views/find-symbols-view');
+
+interface TemporaryCodeAction {
```
Comment (nosami) [inline lib/omnisharp-atom/features/find-symbols.ts]: Not used

File: lib/omnisharp-atom/features/find-symbols.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import Omni = require('../../omni-sharp-server/omni')
+import ClientManager = require('../../omni-sharp-server/client-manager');
+import OmniSharpAtom = require('../omnisharp-atom')
+import SpacePen = require('atom-space-pen-views');
```
Comment (nosami) [inline lib/omnisharp-atom/features/find-symbols.ts]: not used

File: lib/omnisharp-atom/features/find-symbols.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import Omni = require('../../omni-sharp-server/omni')
+import ClientManager = require('../../omni-sharp-server/client-manager');
```
Comment (nosami) [inline lib/omnisharp-atom/features/find-symbols.ts]: Not used

File: lib/omnisharp-atom/features/find-symbols.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+    private view : FindSymbolsView;
+    private editor : Atom.TextEditor;
+
+    constructor(private atomSharper: typeof OmniSharpAtom) {
+        this.atomSharper = atomSharper;
+    }
+
+    public activate() {
+        this.atomSharper.addCommand("omnisharp-atom:find-symbols", () => {
```
Comment (nosami) [inline lib/omnisharp-atom/features/find-symbols.ts]: Use OmniSharpAtom.addCommand instead. It handles making sure that the command is only fired on C# files.

File: lib/omnisharp-atom/views/find-symbols-view.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
+import OmniSelectListView = require('../features/lib/omni-select-list-view');
+import ClientManager = require('../../omni-sharp-server/client-manager');
+import Omni = require('../../omni-sharp-server/omni');
+import $ = require('jquery');
```
Comment (nosami) [inline lib/omnisharp-atom/views/find-symbols-view.ts]: not used

File: lib/omnisharp-atom/features/lib/omni-select-list-view.ts [hunk 7]
```diff
@@ -0,0 +1,65 @@
+import SpacePen = require('atom-space-pen-views');
+import ClientManager = require('../../../omni-sharp-server/client-manager');
+import Omni = require('../../../omni-sharp-server/omni');
+import $ = require('jquery');
```
Comment (nosami) [inline lib/omnisharp-atom/features/lib/omni-select-list-view.ts]: not used

File: lib/omnisharp-atom/features/lib/omni-select-list-view.ts [hunk 8]
```diff
@@ -0,0 +1,65 @@
+import SpacePen = require('atom-space-pen-views');
+import ClientManager = require('../../../omni-sharp-server/client-manager');
+import Omni = require('../../../omni-sharp-server/omni');
```
Comment (nosami) [inline lib/omnisharp-atom/features/lib/omni-select-list-view.ts]: not used

File: lib/omnisharp-atom/features/lib/omni-select-list-view.ts [hunk 9]
```diff
@@ -0,0 +1,65 @@
+import SpacePen = require('atom-space-pen-views');
+import ClientManager = require('../../../omni-sharp-server/client-manager');
```
Comment (nosami) [inline lib/omnisharp-atom/features/lib/omni-select-list-view.ts]: not used

File: lib/omnisharp-atom/features/find-symbols.ts (hunk 5 above)
Comment (nosami) [inline lib/omnisharp-atom/features/find-symbols.ts]: You can remove the constructor then too.

**Comment by nosami** [comment]
Looks cool though. I wonder if we should apply a limit on the number of results displayed?

File: keymaps/omnisharp-atom.cson (hunk 1 above)
Comment (nosami) [inline keymaps/omnisharp-atom.cson]: You can also press `cmd-.` then a key to see if it's used. 

We should also add `ctrl-,` for non resharper users :)

Comment (nosami) [inline keymaps/omnisharp-atom.cson]: readme needs updating too.

Comment (ctolkien) [inline keymaps/omnisharp-atom.cson]: >You can also press cmd-. then a key to see if it's used.

Not anymore, we've hijacked it for code actions. You can activate the Packages -> Keybinding Resolver -> Toggle menu.

Comment (nosami) [inline keymaps/omnisharp-atom.cson]: @ctolkien that's `ctrl-.` :)

Comment (ctolkien) [inline keymaps/omnisharp-atom.cson]: @nosami  - is that not the same as ctrl on mac? I can't keep up with all the modifier keys they've added (but you only need one mouse button!).

Comment (nosami) [inline keymaps/omnisharp-atom.cson]: No.. ctrl and cmd are separate modifiers

**Comment by mat-mcloughlin** [comment]
Cool, I've made the changes and will push it at lunch. @nosami should the appveyor build be failing?

**Comment by nosami** [comment]
Looks like an issue with Atom.IO. Probably nothing to worry about. Yeah, 50 sounds like a good limit.

**Comment by mat-mcloughlin** [comment]
:+1:

**Comment by nosami** [comment]
Looks good. Cheers!

**Pull request merged**